lib/aerospike/client.rb

```diff
@@ -894,13 +894,12 @@ def execute_batch_index_commands(keys)
 
       batch_nodes.each do |batch|
         threads << Thread.new do
-          Thread.current.abort_on_exception = true
           command = yield batch.node, batch
           execute_command(command)
         end
       end
 
-      threads.each(&:join)
+      threads.each(&:value)
```

Comment:
You are right, we don't need to change this

Reply:
Just as a comment, since @madejejej removed that while I was writing a comment:

@madejejej correct me if I'm wrong, but this is the last invocation of `#execute_batch_index_commands`, which means that when `abort_on_exception` is set to false, the outcome of it when running with `#value` will be either the threads enumerator / threads pool in case all went well or it will rerase the first stumbled uppon exception (ref: https://ruby-doc.org/core-2.5.0/Thread.html#method-i-value).

However I do see a design flaw here. If we have several errors from different dead threads,  only one will be propagated. This does not mean that it will be the first one (root cause).